### DISCUSSION TITLE
Unable to add multiple interfaces in cable connections through API

### DISCUSSION THREAD

**Discussion by trs91km** [netbox-community/netbox#12830] [Q&A] [answered]
I am having bonding on device A with multiple interfaces on device B, if I try to make cable connections am getting error like Device A interface is already connected, how to fix this?
my below sample CSV
side_a_device,side_a_type,side_a_name,side_b_device,side_b_type,side_b_name
[device1.com](http://device1.com/),dcim.interface,eth1,[device2.com](http://device2.com/),dcim.interface,xe-1-0
[device1.com](http://device1.com/),dcim.interface,eth1,[device2.com](http://device2.com/),dcim.interface,xe-1-1

I tried to alter csv like below as well, doesn't help
side_a_device,side_a_type,side_a_name,side_b_device,side_b_type,side_b_name
device1.com,dcim.interface,eth1,[device2.com](http://device2.com/),dcim.interface,(xe-1-0,xe-1-1)-> in single column

We have option in GUI to add multiple interfaces but through API its failing

**Comment by candlerb** [comment]
CSV != API

You can search in the previous discussions, but from memory, CSV doesn't support multiple terminations, but the REST API does.

**Comment by trs91km** [comment]
Am using API only, just reading the content from CSV

  #  response=$(curl -s -H "Authorization: Token $NETBOX_API_TOKEN" -H "Content-Type: application/json" -X POST -d '{
     24     "termination_a_type": "dcim.interface",
     25     "termination_a_id": "'"$termination_a_interface_id"'",
     26     "termination_b_type": "dcim.interface",
     27     "termination_b_id": "'"$termination_b_interface_id"'",
     28     "status": "connected",
     29     "type": "'"$cable_type"'"
     30   }' "$NETBOX_API_URL/dcim/cables/")

**Reply by candlerb** [reply]
You didn't say what version of Netbox you're using, but recent versions of Netbox have an array of terminations at each end, not a single termination_a and termination_b.  I'm pretty sure this was discussed fairly recently, so you should find some working examples by searching the discussion history.

**Reply by trs91km** [reply]
Am using version 3.52, sure will go through the discussions, if you have the link handy please share it. Thanks

**Reply by trs91km** [reply, marked as answer]
Finally found the API commands to add multiple interfaces from termination B to termination A



 curl -v -H "Authorization: Token $TOKEN" -H "Content-Type: application/json" -X POST -d ' { 
"a_terminations": [
                {
                    "object_type": "dcim.interface",
					"object_id": 60,
                    "object": {
                        "device": {
                            "name": "device2.site1.bitgravity.com"
                        },
                        "name": "eth1"
                    }
                }
            ],
"b_terminations": [
                {
                    "object_type": "dcim.interface",
					"object_id": 58,
                    "object": {
                        "device": {

                            "name": "device3.site1.bitgravity.com"
                        },
                        "name": "xe-1-1"
                    }
                },
                {
                    "object_type": "dcim.interface",
					"object_id": 57,
                    "object": {
                        "device": {

                            "name": "device3.site1.bitgravity.com"
                        },
                        "name": "xe-1-0"
                    }
                }				
            ],
            "status": "connected"
            
}' "http://$URL/api/dcim/cables/"
